src/Raven.Server/Documents/Includes/IncludeCompareExchangeValuesCommand.cs

```diff
@@ -109,12 +110,12 @@ internal void Materialize(long maxAllowedRaftId)
                 if (string.IsNullOrEmpty(includedKey))
                     continue;
 
-                if (TryGetAtomicGuard(includedKey, maxAllowedRaftId, out var index, out var value) == false)
-                    continue;
-                
-                Results ??= new Dictionary<string, CompareExchangeValue<BlittableJsonReaderObject>>(StringComparer.OrdinalIgnoreCase);
+                var toAdd = TryGetCompareExchange(includedKey, maxAllowedAtomicGuardIndex, out var index, out var value)
+                    ? (index, value)
+                    : (-1, null);
 
-                Results.Add(includedKey, new CompareExchangeValue<BlittableJsonReaderObject>(includedKey, index, value));
+                Results ??= new Dictionary<string, CompareExchangeValue<BlittableJsonReaderObject>>(StringComparer.OrdinalIgnoreCase);
+                Results.Add(includedKey, new CompareExchangeValue<BlittableJsonReaderObject>(includedKey, toAdd.index,  toAdd.value));
```

Comment:
do we really send here `-1` and `null`? what does it mean?

Reply:
It means the compare exchange is missing

Reply:
Yes
https://github.com/ravendb/ravendb/pull/18503/files#r1596990992

Reply:
okay, so what will happened if old client is talking to a new server?

Reply:
> Explicit handling missing compare exchange but the behavior is the same if we call here RegisterCompareExchangeValue instead (Except that CompareExchangeSessionValue._originalValue will be set).
So no worries if the client is not updated

The same.
It will mark it as deleted